website/docs: expand nginx reverse-proxy setup

sevmonster opened the pull request:
## Changes
Clean up the reverse-proxy documentation to provide a bit more information and rationale. The original nginx configuration was also not able to use keepalive connections to authentik since the `Connection: close` header would more than likely be sent with every request that wasn't a WebSocket handshake, since `$http_upgrade` will be empty most of the time.

## Additional
Could cut down on the wordiness but I wanted to be verbose.

sevmonster committed:
website/docs: expand nginx reverse-proxy setup

BeryJu commented on website/docs/installation/reverse-proxy.md:
```diff
@@ -2,40 +2,75 @@
+    }
+
+    # WebSocket
+    location /ws/client/ {
```
This should be `/ws/` since there's another websocket endpoint besides `/ws/client/`, namely `/ws/outpost/<uuid>/` for Outposts (also I don't get the point of having a separate location block for websockets compared to just setting the headers correctly in the main block)

sevmonster replied:
I could only see `client` when glancing at the code/page requests, thanks.
I don't know why I didn't think of it before, but you could just rewrite the mapping to replace `close` with the empty string, and both blocks could be combined. This is just how I have been doing it in all my configs, since I never declared my own mapping and used the mapping from `nginx.conf` provided by the package maintainer. I will change it.
My previous train of thought was that on most configurations and with the current setup, the `$connection_upgrade` mapping is `close` for all other requests than ones with the `Upgrade` header set, so merging them into one would mean that those requests would not be persistent.

sevmonster committed:
website/docs: simplify reverse-proxy config

Combine location blocks and remove documentstion links, they were not doing much.

sevmonster committed:
Merge branch 'goauthentik:main' into patch-1

sevmonster commented:
@BeryJu how does it look now?
